xarray/core/variable.py

```diff
@@ -1095,64 +919,11 @@ def _replace(
         attrs=_default,
         encoding=_default,
     ) -> T_Variable:
-        if dims is _default:
-            dims = copy.copy(self._dims)
-        if data is _default:
-            data = copy.copy(self.data)
-        if attrs is _default:
-            attrs = copy.copy(self._attrs)
+        new_object = super()._replace(dims, data, attrs)
```

Comment:
@dcherian, regarding the suggestion in https://github.com/pydata/xarray/pull/8075#discussion_r1300780051, duplicating the sentinel value _default breaks the instantiation of NamedArray instances. one place where this issue arises is in the `._replace()` method:


```python
   def _replace(self, dims=_default, data=_default, attrs=_default):
        if dims is _default:
            dims = copy.copy(self._dims)
        if data is _default:
            data = copy.copy(self._data)
        if attrs is _default:
            attrs = copy.copy(self._attrs)
        return type(self)(dims, data, attrs)
```

this method is crucial and `... is _default` check is why i initially chose to define the sentinel value in one place and reuse it instead of duplicating it.


```python

/home/andersy005/devel/pydata/xarray/xarray/tests/test_variable.py:2849: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/home/andersy005/devel/pydata/xarray/xarray/core/variable.py:1074: in as_numpy
    return self._replace(data=self.to_numpy())
/home/andersy005/devel/pydata/xarray/xarray/core/variable.py:922: in _replace
    new_object = super()._replace(dims, data, attrs)
/home/andersy005/devel/pydata/xarray/xarray/namedarray/core.py:275: in _replace
    return type(self)(dims, data, attrs)
/home/andersy005/devel/pydata/xarray/xarray/core/variable.py:2634: in __init__
    super().__init__(dims, data, attrs, encoding, fastpath)
/home/andersy005/devel/pydata/xarray/xarray/core/variable.py:349: in __init__
    super().__init__(
/home/andersy005/devel/pydata/xarray/xarray/namedarray/core.py:50: in __init__
    self._dims: tuple[Hashable, ...] = self._parse_dimensions(dims)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <[AttributeError("'IndexVariable' object has no attribute '_dims'") raised in repr()] IndexVariable object at 0x7fa57a690220>
dims = <Default.token: 0>

    def _parse_dimensions(self, dims: str | Iterable[Hashable]) -> tuple[Hashable, ...]:
>       dims = (dims,) if isinstance(dims, str) else tuple(dims)
E       TypeError: 'Default' object is not iterable
```

Reply:
found the culprit bug that was causing this. it should be fixed now